src/install/npm_package_arg.zig

```diff
@@ -0,0 +1,1711 @@
+            .raw = raw_slice,
+            .name = name_slice,
+            .raw_spec = raw_spec_slice,
+            .#arena_buffer = arena.buffer,
+            .#fetch_spec_slice = null,
+            .save_spec = null,
+            .type = .{
+                .alias = .{
+                    .sub_spec = sub_spec_ptr,
+                },
+            },
+            .#allocator = allocator,
+        };
+    }
+
+    fn fromGitSpec(
+        allocator: std.mem.Allocator,
+        name: ?[]const u8,
+        raw_spec: []const u8,
+        raw_arg: ?[]const u8,
+    ) !?NpaSpec {
+        const mut_spec_str: []u8 = try allocator.dupe(u8, raw_spec);
+        defer allocator.free(mut_spec_str);
+
+        var hosted = try HostedGitInfo.fromUrl(allocator, mut_spec_str) orelse {
+            return null;
+        };
+        errdefer hosted.deinit();
+
+        var git_attrs = if (hosted.committish) |c|
+            try GitAttrs.fromCommittish(allocator, c)
+        else
+            null;
+        errdefer if (git_attrs) |*g| g.deinit();
+
+        const specs = try hosted.toStringBoth(allocator);
+        defer allocator.free(specs.save_spec);
+        defer if (specs.fetch_spec) |f| allocator.free(f);
+
+        var arena_size: usize = rawFieldLength(raw_arg, name, raw_spec);
+        arena_size += mut_spec_str.len + specs.save_spec.len;
+        if (name) |n| arena_size += n.len;
+        if (specs.fetch_spec) |f| arena_size += f.len;
+
+        const buffer = try allocator.alloc(u8, arena_size);
+        errdefer allocator.free(buffer);
+
+        var arena = StringArena.init(buffer);
+
+        const raw_slice = arena.copyRawField(raw_arg, name, raw_spec);
+        const name_slice = arena.maybeCopy(name);
+        const raw_spec_slice = arena.copyString(mut_spec_str);
+        const save_spec_slice = arena.copyString(specs.save_spec);
+        const fetch_spec_slice = if (specs.fetch_spec) |f|
+            arena.copyString(f)
+        else
+            null;
+
+        return .{
+            .raw = raw_slice,
+            .name = name_slice,
+            .raw_spec = raw_spec_slice,
+            .#arena_buffer = arena.buffer,
+            .#fetch_spec_slice = fetch_spec_slice,
+            .save_spec = save_spec_slice,
+            .type = .{
+                .git = .{
+                    .attrs = git_attrs,
+                    .hosted = hosted,
+                },
+            },
+            .#allocator = allocator,
+        };
+    }
+
+    fn fromFile(
+        allocator: std.mem.Allocator,
+        name: ?[]const u8,
+        raw_spec: []const u8,
+        where: []const u8,
+        raw_arg: ?[]const u8,
+    ) !Self {
+        const stack_fallback_bytes = 8192;
+
+        var stack_fallback = std.heap.stackFallback(stack_fallback_bytes, allocator);
+        const temp = stack_fallback.get();
+
+        var raw_spec_cleaned = PathToFileUrlUtils.cleanPathToFileUrl(temp, raw_spec) catch {
+            return error.InvalidPath;
+        };
```

Comment:
cleanPathToFileUrl only returns an error on OOM, but it's turned into error.InvalidPath here. should this instead be `try cleanPathToFileUrl`?